W-15439366 Update flex-troubleshoot-requests.adoc (#546)
mulesoft/docs-gateway: gateway/1.7/modules/ROOT/pages/flex-troubleshoot-platform-connection.adoc

> = Troubleshooting Platform Connection

Reviewer: ```suggestion
= Troubleshooting Platform Connections
```

> = Troubleshooting Platform Connections

Reviewer: This should match the TOC: Troubleshooting > Troubleshoot

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> When configuring Flex Gateway, it is possible for the Flex Replica's `registration.yaml` to be absent or for the replica to lose connection to Anypoint Platform.

Reviewer: ```suggestion
When configuring Flex Gateway, it is possible for the Flex Replica's `registration.yaml` to be absent or for the replica to lose its connection to Anypoint Platform.
```

> When configuring Flex Gateway in either mode, it is possible for the Flex Replica's `registration.yaml` to not be present or for the replica to lose connection to Anypoint Platform.

Reviewer: Do you need to say "in either mode"? I don't think it's necessary.

Instead of "to not be present," I suggest "to be absent."

> To troubleshoot if a registration file is present and if Flex Gateway can make platform connections, Flex Gateway provides the following command:

Reviewer: ```suggestion
To troubleshoot whether a registration file is present and whether Flex Gateway can make platform connections, use the following command:
```

[source,ssh]
----
flexctl check connections
----

> The `flexctl check connections` first ensures a `registration.yaml` file is present, and then checks if the replica can connect to the URLs defined in the registration file. If the command is successful, it returns the following `info` message:

Reviewer: ```suggestion
The `flexctl check connections` command first ensures that a `registration.yaml` file is present. Then it checks whether the replica can connect to the URLs defined in the registration file. If the `flexctl check connections` command is successful, it returns the following `info` message:
```

----
$ flectl check connections
[flexctl][info] Connection to platform endpoints successful.
----

== No Registration File Found

> If the `registration.yaml` file is not found, Flex Gateway returns the following error message:

Reviewer: ```suggestion
If the `registration.yaml` file isn't found, Flex Gateway returns the following error message:
```

----
$ flexctl check connections
Error: no registration config
----

To troubleshoot an absent registration file, check your Flex Replica configuration folder. To learn more about registering a Flex Gateway, see following section for your relevant mode:

* xref:flex-conn-reg-run.adoc[]
* xref:flex-local-reg-run.adoc[]

== Failed Connection

> There are multiple reasons Flex Gateway may experience connection issues, such as an error in the gateway's environment or a network issue. How you troubleshoot your error is dependent on the error returned.

Reviewer: ```suggestion
There are multiple reasons why Flex Gateway might experience connection issues, such as an error in the gateway's environment or a network issue. How you troubleshoot your error depends on the error returned.
```

> If Flex Gateway cannot to a specific URL, it returns an error message that includes the URL that was unreachable and the error message returned from the connection attempt. The error message appears similar to the following:

Reviewer 2: 1. missing word `If Flex Gateway cannot **connect** to a specific URL`
2. I'm not sure about `The error message appears similar to the following`, since it may not really look like that. I would say something like `For example, this is an error shown when the platform is unreachable through the gateway because of no internet connection`

> If Flex Gateway cannot connect to a specific URL, it returns an error message that includes the URL that was unreachable and the error message returned from the connection attempt. The following error message is an example of an error due to no internet connection:

Reviewer: ```suggestion
If Flex Gateway can't connect to a specific URL, it returns an error message that includes the unreachable URL and the error message returned from the connection attempt. The following error message is an example of an error returned due to no internet connection:
```

[source,ssh]
----
flexctl check connections
----

The `flexctl check connections` first ensures a `registration.yaml` file is present, and then checks if the replica can connect to the URLs defined in the registration file. If the command is successful, it returns the following `info` message:

----
$ flectl check connections
[flexctl][info] Connection to platform endpoints successful.
----

== No Registration File Found

If the `registration.yaml` file is not found, Flex Gateway returns the following error message:

----
$ flexctl check connections
Error: no registration config
----

To troubleshoot an absent registration file, check your Flex Replica configuration folder. To learn more about registering a Flex Gateway, see following section for your relevant mode:

* xref:flex-conn-reg-run.adoc[]
* xref:flex-local-reg-run.adoc[]

== Failed Connection

If Flex Gateway cannot to a specific URL, it returns the following error message and the status code of the connection request:

----
$ flexctl check connections
Error: failed connecting to https://arm-mcm2-service.kstg.msap.io.
Status: 404
Response: <html><body><h1>Resource not found</h1></body></html>

> ----

Reviewer 2: I'm not sure about including this since it's just a test error, not really going to get something like this. I would replace it with a network error like this one, but being very clear that it is an example
```
$ flexctl check connections
Error: failed connecting to https://anypoint.mulesoft.com/: dial tcp 52.22.227.22:443: connect: network is unreachable
```

> How you troubleshoot a failed connection is dependent on what connection failed and what status code was returned.

Reviewer 2: I would say that it is dependent on the error outputted by the command, not really the status code. For instance, the example network error I posted above is an indication that there is an issue on the gateways environment that makes the platform unreachable
